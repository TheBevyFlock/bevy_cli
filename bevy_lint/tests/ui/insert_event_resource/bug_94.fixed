//! [#94]: https://github.com/TheBevyFlock/bevy_cli/issues/94

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_event_resource)]

use bevy::prelude::*;

#[derive(Event)]
struct Foo;

fn main() {
    let mut app = App::new();

    // These both should error, but currently do not.
    App::add_event::<Foo>(&mut app);
    //~^ ERROR: called `App::init_resource::<Events<Foo>>(&mut app)` instead of
    // `App::add_event::<Foo>(&mut app)`
    App::add_event::<Foo>(&mut app);
    //~^ ERROR: called `App::insert_resource::<Events<Foo>>(Default::default())` instead of
    // `App::add_event::<Foo>(&mut app)`
}
