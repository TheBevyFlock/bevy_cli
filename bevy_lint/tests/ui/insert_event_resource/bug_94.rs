//! This test tracks the bug reported in [#94]. When this starts failing, the bug has been fixed.
//!
//! [#94]: https://github.com/TheBevyFlock/bevy_cli/issues/94

//@check-pass

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_event_resource)]

use bevy::prelude::*;

#[derive(Event)]
struct Foo;

fn main() {
    let mut app = App::new();

    // These both should error, but currently do not.
    App::init_resource::<Events<Foo>>(&mut app);
    App::insert_resource::<Events<Foo>>(&mut app, Default::default());
}
