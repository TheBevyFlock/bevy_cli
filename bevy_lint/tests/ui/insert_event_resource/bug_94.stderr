error: called `App::init_resource::<Events<Foo>>(&mut app)` instead of `App::add_event::<Foo>(&mut app)`
  --> tests/ui/insert_event_resource/bug_94.rs:16:5
   |
16 |     App::init_resource::<Events<Foo>>(&mut app);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/insert_event_resource/bug_94.rs:5:9
   |
5  | #![deny(bevy::insert_event_resource)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: inserting an `Events` resource does not fully setup that event
   |
16 |     App::add_event::<Foo>(&mut app);
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: called `App::insert_resource::<Events<Foo>>(Default::default())` instead of `App::add_event::<Foo>(&mut app)`
  --> tests/ui/insert_event_resource/bug_94.rs:19:5
   |
19 |     App::insert_resource::<Events<Foo>>(&mut app, Default::default());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Events` resource does not fully setup that event
   |
19 |     App::add_event::<Foo>(&mut app);
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to 2 previous errors

